src/bank.rs

```diff
@@ -637,6 +638,14 @@ impl Bank {
         false
     }
 
+    pub fn hash_bank_state(&self) -> Hash {
+        let mut ordered_accounts = BTreeMap::new();
```

Comment:
At a glance, it looks like it ought to be possible to write this like:

```rust
let ordered_accounts: BTreeMap<_> = self.accounts.read().unwrap().iter().cloned().collect();
```

Is it?

Reply:
Sure seems like it ought to be possible, but I haven't found a way. This produces a type mismatch error. The Hashmap iterator element type is `(&K, &V)` and .cloned() expects `&T`

Reply:
Right, so instead of `cloned()`, you'd use `map(|(k, v)| (k.clone(), v.clone()))`. Given that, what you've written is probably clearer.